src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -1455,26 +1501,42 @@ private String translateFactExpression(String expression) {
     }
 
     public int getConsumerEntitlementCount(Owner owner, ConsumerType type) {
-        Criteria c = createSecureCriteria()
-            .add(Restrictions.eq("ownerId", owner.getId()))
-            .add(Restrictions.eq("typeId", type.getId()))
-            .createAlias("entitlements", "ent")
-            .setProjection(Projections.sum("ent.quantity"));
-
-        Long result = (Long) c.uniqueResult();
-        return result == null ? 0 : result.intValue();
+        EntityManager em = getEntityManager();
+        CriteriaBuilder cb = em.getCriteriaBuilder();
+
+        CriteriaQuery<Integer> cq = cb.createQuery(Integer.class);
+        Root<Consumer> consumerRoot = cq.from(Consumer.class);
+
+        Join<Consumer, Entitlement> entitlementJoin = consumerRoot.join("entitlements");
+
+        Predicate securePredicate = getSecurityPredicate(Consumer.class, cb, consumerRoot);
+        if (securePredicate == null) {
+            securePredicate = cb.conjunction();
+        }
+
+        Predicate ownerPredicate = cb.equal(consumerRoot.get("ownerId"), owner.getId());
+        Predicate typePredicate = cb.equal(consumerRoot.get("typeId"), type.getId());
+
+        cq.select(cb.sum(entitlementJoin.get("quantity")))
+            .where(cb.and(ownerPredicate, typePredicate, securePredicate));
+
+        Integer result = em.createQuery(cq).getSingleResult();
+        return result != null ? result : 0;
     }
 
-    @SuppressWarnings("unchecked")
     public List<String> getConsumerIdsWithStartedEnts() {
         Date now = new Date();
-        return currentSession().createCriteria(Entitlement.class)
-            .createAlias("pool", "p")
-            .add(Restrictions.eq("updatedOnStart", false))
-            .add(Restrictions.lt("p.startDate", now))
-            .setProjection(Projections.property("consumer.id"))
-            .setResultTransformer(Criteria.DISTINCT_ROOT_ENTITY)
-            .list();
+        String jpql = """
+            SELECT DISTINCT e.consumer.id FROM Entitlement e
+            JOIN e.pool p
+            WHERE e.updatedOnStart = FALSE
+                AND p.startDate < :now
```

Comment:
We need not pass the current date in as a parameter in this case. Instead we can use `NOW()`, `CURRENT_TIMESTAMP`, `CURRENT_DATE`, or `CURRENT_TIME` in our queries as appropriate.